### ISSUE TITLE
speed up bedboss / bedboss profiling

### ISSUE THREAD

**Issue by khoroshevskyi** [databio/bedboss#78]
I ran bedboss profiling, and got this results:

![image](https://github.com/user-attachments/assets/00413e5c-7d0b-427b-9c84-378e8a6e03fd)

Summary:
1) **Bedstats** -> 44s.
2) **Uploading files** -> 8.4 s.
3) **Initialization of bbcon**f -> 7.2s. (Is initialized once, before running multiple files. - So it's not a problem)
4) **Bedmaker** -> 0.6s.
5) **Genome validator** -> 0.08 s.
6) **Bedqc** -> 0.081

In conclusion, the most time-consuming part is the calculation of bed statistics. Every other part of the pipeline is relatively fast, but on Rivanna, it is even slower (though I’m not sure why).

**Comment by nsheff** [comment]
Maybe related to #89

**Comment by khoroshevskyi** [comment]
A lot of changes has been done to bedboss, including: RegionSet now is written in Rust, processing was divided into heavy and lite processing. I close this issue for now. The slowest part is still R stats calculation.
